nss/src/cache/mod.rs

```diff
@@ -300,7 +304,15 @@ impl CacheDBBuilder {
 
         let passwd_db = &db_path.join(PASSWD_DB);
         let passwd_db = passwd_db.to_str().unwrap();
-        let conn = match Connection::open_with_flags(passwd_db, open_flags) {
+    
+        let mut passwd_db_args = DB_CON_OPT_RO;
+        if shadow_mode == ShadowMode::ReadWrite { 
+            passwd_db_args = DB_CON_OPT_RW;
+        }
+        let passwd_db_con = format!("{}{}{}", DB_CON_PREFIX.to_string(), passwd_db, passwd_db_args);
```

Comment:
There's no need to call `.to_string()` in a &str type when formatting or printing it.

Reply:
Hello, thanks for the feedback. fixed in latest commit. :)